src/main/java/com/codeborne/selenide/conditions/TextCondition.java

```diff
@@ -0,0 +1,42 @@
+package com.codeborne.selenide.conditions;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.Driver;
+import org.openqa.selenium.WebElement;
+
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+@ParametersAreNonnullByDefault
+public abstract class TextCondition extends Condition {
+  private final String expectedText;
+  private final ThreadLocal<String> actualText = new ThreadLocal<>();
```

Comment:
@asolntsev jfyi, I get `"ThreadLocal" variables should be cleaned up when no longer used` sonar rule violation locally. I don't know exactly it is important for us or no